src/render/painter.js

```diff
@@ -477,7 +502,12 @@ class Painter {
         this.context.viewport.set([0, 0, this.width, this.height]);
 
         // Clear buffers in preparation for drawing to the main framebuffer
-        this.context.clear({color: options.showOverdrawInspector ? Color.black : Color.transparent, depth: 1});
+        // If fog is enabled, use the fog color as default clear color.
+        let clearColor = Color.transparent;
+        if (this.style.fog) {
+            clearColor = this.style.fog.properties.get('color');
```

Comment:
Are we sure we want to clear to the fog color? Would this ever be jarring in cases where there is a very small amount of fog near the horizon?

Reply:
That's a good question. I think that's important as leaving transparent fragments in our viewport with unassigned colors is a bit of an issue at high pitch since we're dependent on whatever is drawn previously. Using fog color as clear color ensures consistency between that empty rendered space and our rendered scene.

But do you think that to be too intrusive? The case I could think of where this can be an issue is having a scene composed of only transparent layers. I'm not against leaving the background transparent and it would be up to the user to make sure the scene is consistent.

A bad reference case between leaving a transparent clear (with a mismatching css background color set to black) vs with using the clear color as fog color:

https://user-images.githubusercontent.com/7061573/115628007-c81ba500-a2b4-11eb-9da8-a69d175a8a1d.mov

https://user-images.githubusercontent.com/7061573/115628024-ceaa1c80-a2b4-11eb-8190-daa6fe4e92b3.mov

Reply:
I was leaning towards removing this, and I investigated this removal in https://github.com/mapbox/mapbox-gl-js/commit/5ef099db3422aca72189b1ad7abf1766a7131d40. My conclusions are that we should leave it in:
- Culling might lead to even more empty space, as noted in the rebaseline render tests removing the clear color as fog color:
<img width="773" alt="Screen Shot 2021-04-23 at 1 10 26 PM" src="https://user-images.githubusercontent.com/7061573/115925468-012f5300-a436-11eb-90b8-2ed836b222ab.png">
<img width="803" alt="Screen Shot 2021-04-23 at 1 09 21 PM" src="https://user-images.githubusercontent.com/7061573/115925811-9e8a8700-a436-11eb-96f0-35d4425130b9.png">

- For the reasons mentioned in my previous comment, leaving empty rendered space can lead to incoherence in the scene over the horizon or when we have this potential cutoff which may look like rendering artifacts